agent/hcp/client/http_client_test.go

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package client
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/hashicorp/consul/agent/hcp/config"
+	"github.com/hashicorp/go-hclog"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNewHTTPClient(t *testing.T) {
+	mockCfg := config.MockCloudCfg{}
+	mockHCPCfg, err := mockCfg.HCPConfig()
+	require.NoError(t, err)
+
+	client := NewHTTPClient(mockHCPCfg.APITLSConfig(), mockHCPCfg, hclog.NewNullLogger())
+	require.NotNil(t, client)
+
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		require.Equal(t, "Bearer test-token", r.Header.Get("Authorization"))
+	}))
+	client.Get(srv.URL)
```

Comment:
I feel this test could be provide a little more confidence if the assertion ran unconditionally — as in, not inside a function that could potentially not be called. What do you think about something like this?
```suggestion
	var req *http.Request
	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		req = r
	}))
	client.Get(srv.URL)
	require.Equal(t, "Bearer test-token", req.Header.Get("Authorization"))
```

Reply:
That's a good point, will make this test improvement!